databricks/koalas/frame.py

```diff
@@ -9659,6 +9657,118 @@ class  max_speed
         elif axis == 1:
             return self.iloc[:, indices]
 
+    def eval(self, expr, inplace=False):
+        """
+        Evaluate a string describing operations on DataFrame columns.
+
+        Operates on columns only, not specific rows or elements. This allows
+        `eval` to run arbitrary code, which can make you vulnerable to code
+        injection if you pass user input to this function.
+
+        Parameters
+        ----------
+        expr : str
+            The expression string to evaluate.
+        inplace : bool, default False
+            If the expression contains an assignment, whether to perform the
+            operation inplace and mutate the existing DataFrame. Otherwise,
+            a new DataFrame is returned.
+
+        Returns
+        -------
+        The result of the evaluation.
+
+        See Also
+        --------
+        DataFrame.query : Evaluates a boolean expression to query the columns
+            of a frame.
+        DataFrame.assign : Can evaluate an expression or function to create new
+            values for a column.
+        eval : Evaluate a Python expression as a string using various
+            backends.
+
+        Examples
+        --------
+        >>> df = ks.DataFrame({'A': range(1, 6), 'B': range(10, 0, -2)})
+        >>> df
+           A   B
+        0  1  10
+        1  2   8
+        2  3   6
+        3  4   4
+        4  5   2
+        >>> df.eval('A + B')
+        0    11
+        1    10
+        2     9
+        3     8
+        4     7
+        Name: 0, dtype: int64
+
+        Assignment is allowed though by default the original DataFrame is not
+        modified.
+
+        >>> df.eval('C = A + B')
+           A   B   C
+        0  1  10  11
+        1  2   8  10
+        2  3   6   9
+        3  4   4   8
+        4  5   2   7
+        >>> df
+           A   B
+        0  1  10
+        1  2   8
+        2  3   6
+        3  4   4
+        4  5   2
+
+        Use ``inplace=True`` to modify the original DataFrame.
+
+        >>> df.eval('C = A + B', inplace=True)
+        >>> df
+           A   B   C
+        0  1  10  11
+        1  2   8  10
+        2  3   6   9
+        3  4   4   8
+        4  5   2   7
+        """
+        if isinstance(self.columns, pd.MultiIndex):
+            raise ValueError("`eval` is not supported for multi-index columns")
+        inplace = validate_bool_kwarg(inplace, "inplace")
+        is_series = False
+        is_scalar_ = False
+
+        # TODO: Cannot operate `inplace=True` if there is not assignment in `expr`.
+        # There is no way to check now, so leave it to pandas for now with minimum data.
+        if inplace:
+            self.head(0).to_pandas().eval(expr, inplace)
+
+        def eval_func(pdf):
+            nonlocal is_series
+            nonlocal is_scalar_
```

Comment:
The variable doesn't work here because the function is serielaised to another node. It's difficult to have one global status for all other nodes. We should infer this from either type hint or schema inference from executing it against small sample of the current DataFrame.

Reply:
I got it, let me fix. Thanks! :)

Reply:
I just inspected about this !
These variables are needed for after running `map_in_pandas` to make proper result matched with `pd.DataFrame.eval()`, not for type inferring
The type inferring you mentioned is performed in the `map_in_pandas` automatically since we're utilizing it.

In short, I think we still need these variables here to post-managing.

Reply:
@HyukjinKwon Interestingly now it works since the function `eval_func` here doesn't have a type-hint and it always falls into infer-schema clause in `map_in_pandas` and update the variable in pandas execution locally.

Reply:
Hm.. I see. If we will infer the schema always, it should be fine. @itholic can you clarify this with come comments?

Reply:
@ueshin @HyukjinKwon , Thanks for the comments. 

Okay, I'll add the comments